functions/src/delete-votes-before.ts

```diff
@@ -0,0 +1,88 @@
+import * as functions from "firebase-functions";
+import * as firebase from "firebase-admin";
+
+import { Config, isEnabled, asNumber } from "./config";
+
+/**
+ * TODO: Adapt the function to work with any collection and any condition
+ */
+
+const config = functions.config() as Config;
+const deleteVotesBeforeConfigs = config.features.delete_votes_before;
+const db = firebase.firestore();
+
+const deleteQueryBatch = (
+  store: FirebaseFirestore.Firestore,
+  query: FirebaseFirestore.Query,
+  resolve: (value?: {} | PromiseLike<{}> | undefined) => void,
+  reject: (reason?: any) => void
```

Comment:
This idiom feels weird to me, I've never seen it anywhere else. Could the function return a promise instead?